[Shipping labels] Handle button visibilities on Create shipping label screen

irfano opened the pull request:
<!-- Remember about a good descriptive title. -->

<!-- Id number of the Linear issue this PR addresses, e.g., WOOMOB-373. -->
Part of WOOMOB-284

### Description
<!-- Take the time to write a good summary. Why is it needed? What does it do? When fixing bugs try to avoid just writing “See original issue” – clarify what the problem was and how you’ve fixed it. -->
This
- Hides the purchase button after buying a shipping label for an order with a single shipment.
- Hides the Split Shipment button when there is only a single item that is not purchased.
- Makes the addresses non-editable for purchased shipments.

> [!WARNING]  
> Please review #14136 first

### Steps to reproduce
<!-- Step-by-step testing instructions. For new user flows, consider instead stating the goal of the workflow and see if your PR reviewer can accomplish the workflow without specific steps! -->
1. Go to the Orders tab and select an order.
2. Tap an order with a single item.
3. Tap on Create shipping label.
7. Fill required forms and purchase a shipping label.

### The tests that have been performed
<!-- To give the reviewer an idea of what could be missed in terms of testing -->
Steps above

### Images/gif
<!-- Include before and after images or gifs when appropriate. -->
|Before|After|
|-|-|
|<img src="https://github.com/user-attachments/assets/a93cd723-af96-4e43-95c1-a5c8be9b21c1" width=350>|<img src="https://github.com/user-attachments/assets/51029206-8aee-48f4-a47d-ef08e015981e" width=350>|
|<img src="https://github.com/user-attachments/assets/a77428d7-3bb8-40d0-bec3-b9fce27422a0" width=350>|<img src="https://github.com/user-attachments/assets/9777d6b1-a511-4457-8d32-06a492329e3e" width=350>|

- [x] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

<!-- Pull request guidelines: https://github.com/woocommerce/woocommerce-android/blob/develop/docs/pull-request-guidelines.md -->

irfano committed:
Add flag to check if all shipments are fulfilled

This change introduces a new boolean flag, `allShipmentsFulfilled`, to the `WooShippingLabelDisplay` view state. This flag is true if all shipments in the `shipmentUIList` have been purchased.

irfano committed:
Hide purchase button and "Split shipment" button when all shipments are fulfilled

irfano committed:
Hide split shipment button when only one item is in the order

This commit hides the "Split Shipment" button in the WooShippingLabelCreationScreen when the order contains only a single item, as splitting is not applicable in this scenario.

irfano committed:
Improve split shipment button visibility logic

irfano committed:
Set the address section to be non-editable for purchased shipments

JorgeMucientes commented on WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/wooshippinglabels/WooShippingLabelCreationViewModel.kt:
```diff
@@ -935,8 +935,14 @@ class WooShippingLabelCreationViewModel @Inject constructor(
+            val shouldShowSplitShipmentButton: Boolean
+                get() {
+                    val unpurchasedShipments = shipmentUIList.filterNot { it.purchased }
+                    return unpurchasedShipments.size > 1 || unpurchasedShipments.first().totalItemQuantity > 1
```
Np, could this code `unpurchasedShipments.first()` lead to `NoSuchElementException` if all items are purchased? 
I wonder if the following equivalent code could be safer: 
```kotlin
val shouldShowSplitShipmentButton: Boolean
    get() = shipmentUIList.filterNot { it.purchased }.let { unpurchased ->
        unpurchased.size > 1 || (unpurchased.firstOrNull()?.totalItemQuantity ?: 0) > 1
    }
```

irfano replied:
Good catch! I applied your suggestion: 1fee2c81bb5f3e84594dadd99d5d41bc6c289dd9

JorgeMucientes reviewed (approved):
Code looks good and works as expected. Nicely done @irfano. 
I just left a minor suggestion worth double checking ;)

irfano committed:
Fix NPE when checking for split shipment button visibility

The split shipment button visibility logic could cause a NullPointerException if `unpurchasedShipments` was empty.

irfano committed:
Merge branch 'issue/WOOMOB-284-add-green-print-label-card-to-shipments' into issue/WOOMOB-284-handle-button-visibilities-on-shipping-screen

# Conflicts:
#	WooCommerce/src/main/kotlin/com/woocommerce/android/ui/orders/wooshippinglabels/WooShippingLabelCreationScreen.kt

irfano committed:
Fix detekt error
